Compilation warning: ignoring return value of 'crypto_scalarmult_curve25519'

### The problem

This is a Blitzwolf SHP6 smart socket which has an ESPhome firmware. During compilation I encounter the following warning:

```
/piolibs/noise-c/src/backend/sodium/dh-curve25519.c: In function 'noise_curve25519_calculate':
/piolibs/noise-c/src/backend/sodium/dh-curve25519.c:97:33: warning: ignoring return value of 'crypto_scalarmult_curve25519', declared with attribute warn_unused_result [-Wunused-result]
     crypto_scalarmult_curve25519(
                                 ^
````

The compilation finishes in order and the device seems to work as intended.

### Which version of ESPHome has the issue?

2021.10.1

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

core-2021.10.6

### What platform are you using?

ESP8266

### Board

esp8285

### Component causing the issue

crypto_scalarmult_curve25519

### Example YAML snippet

```yaml
`
substitutions:
  device_name: laptop-charger
  plug_name: laptop_charger
  friendly_name: Laptop charger
  # Higher value gives lower watt readout
  current_res: "0.00221"
  # Lower value gives lower voltage readout
  voltage_div: "920"

esphome:
  name: ${device_name}
  platform: ESP8266
  board: esp8285

wifi:
  ssid: !secret esphome_ssid2
  password: !secret esphome_pwd2
  manual_ip:
    static_ip: 192.168.1.152
    gateway: !secret esphome_gateway
    subnet: !secret esphome_subnet
    dns1: !secret esphome_dns1
    dns2: !secret esphome_dns2
    
  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "${friendly_name} Hotspot"
    password: !secret fallback_password
    
captive_portal:

# Enable logging
logger:

# Enable Web server
web_server:
  port: 80

# Enable Home Assistant API
api:
  password: ''

ota:
  safe_mode: true
  num_attempts: 5
  password: !secret ota_password

time:
  - platform: homeassistant
    id: homeassistant_time
    
binary_sensor:
  - platform: gpio
    internal: true
    pin:
      number: GPIO13
      inverted: True
    on_press:
      - switch.toggle: relay
    name: "${plug_name}_button"
        
switch:
- platform: gpio
  name: "${friendly_name}"
  icon: mdi:power-plug
  id: relay
  pin: GPIO15
  on_turn_on:
    - switch.turn_on: "${plug_name}_LED_Blue"
  on_turn_off:
    - switch.turn_off: "${plug_name}_LED_Blue"
    - switch.turn_off: "${plug_name}_LED_Red"
- platform: gpio
  internal: true
  name: "${plug_name}_LED_Blue"
  id: "${plug_name}_LED_Blue"
  pin: GPIO2
  inverted: True
- platform: gpio
  name: "${plug_name}_LED_Red"
  internal: true
  id: "${plug_name}_LED_Red"
  pin: GPIO0
  inverted: True
sensor:
  - platform: hlw8012
    sel_pin:
      number: GPIO12
      inverted: True
    cf_pin: GPIO05
    cf1_pin: GPIO14
    current_resistor: ${current_res}
    voltage_divider: ${voltage_div}
    current:
      name: "${friendly_name} Current"
      unit_of_measurement: 'A'
      accuracy_decimals: 3
    voltage:
      name: "${friendly_name} Voltage"
      unit_of_measurement: 'V'
    power:
      name: "${friendly_name} Power"
      unit_of_measurement: 'W'
      id: "${plug_name}_Power"
      on_value_range:
        - above: 1.0
          then:
            - switch.turn_on: "${plug_name}_LED_Red"
        - below: 1.0
          then:
            - switch.turn_off: "${plug_name}_LED_Red"
    change_mode_every: 8
    update_interval: 5s
  - platform: total_daily_energy
    name: "${friendly_name} Energy"
    power_id: "${plug_name}_Power"
    filters:
        # Multiplication factor from W to kW is 0.001
        - multiply: 0.001
    unit_of_measurement: kWh

# Extra sensor to keep track of plug uptime
  - platform: uptime
    name: '${friendly_name} Uptime'
    unit_of_measurement: days
    update_interval: 600s
    filters:
      - multiply: 0.0000115741
````
```


### Anything in the logs that might be useful for us?

```txt
This is the compilation setup:

INFO Reading configuration /config/esphome/ironing-plug.yaml...
INFO Detected timezone 'Europe/Budapest'
INFO Generating C++ source...
INFO Compiling app...
Processing ironing-plug (board: esp8285; framework: arduino; platform: platformio/espressif8266 @ 2.6.2)
--------------------------------------------------------------------------------
HARDWARE: ESP8266 80MHz, 80KB RAM, 1MB Flash
PACKAGES: 
 - framework-arduinoespressif8266 3.20704.0 (2.7.4) 
 - tool-esptool 1.413.0 (4.13) 
 - tool-esptoolpy 1.20800.0 (2.8.0) 
 - toolchain-xtensa 2.40802.200502 (4.8.2)
Dependency Graph
|-- <ESPAsyncTCP-esphome> 1.2.3
|   |-- <ESP8266WiFi> 1.0
|-- <ESPAsyncWebServer-esphome> 1.3.0
|   |-- <ESPAsyncTCP-esphome> 1.2.3
|   |   |-- <ESP8266WiFi> 1.0
|   |-- <Hash> 1.0
|   |-- <ESP8266WiFi> 1.0
|   |-- <ArduinoJson-esphomelib> 5.13.3
|-- <ESP8266WiFi> 1.0
|-- <ArduinoJson-esphomelib> 5.13.3
|-- <ESP8266mDNS> 1.2
|   |-- <ESP8266WiFi> 1.0
|-- <noise-c> 0.1.3
|   |-- <libsodium> 1.10018.1
|-- <DNSServer> 1.1.1
|   |-- <ESP8266WiFi> 1.0
```


### Additional information

During compilation I have also this warning: (but this was already reported in #1562 ):

````
In file included from /data/ironing-plug/.piolibdeps/ironing-plug/ESPAsyncWebServer-esphome/src/SPIFFSEditor.cpp:1:0:
/data/ironing-plug/.piolibdeps/ironing-plug/ESPAsyncWebServer-esphome/src/SPIFFSEditor.h:16:101: warning: 'SPIFFS' is deprecated (declared at /data/cache/platformio/packages/framework-arduinoespressif8266/cores/esp8266/FS.h:269): SPIFFS has been deprecated. Please consider moving to LittleFS or other filesystems. [-Wdeprecated-declarations]
     SPIFFSEditor(const String& username=String(), const String& password=String(), const fs::FS& fs=SPIFFS);
````

--- a/esphome/components/api/__init__.py
+++ b/esphome/components/api/__init__.py
@@ -121,7 +121,7 @@ async def to_code(config):
         decoded = base64.b64decode(conf[CONF_KEY])
         cg.add(var.set_noise_psk(list(decoded)))
         cg.add_define("USE_API_NOISE")
-        cg.add_library("esphome/noise-c", "0.1.3")
+        cg.add_library("esphome/noise-c", "0.1.4")
     else:
         cg.add_define("USE_API_PLAINTEXT")
 

--- a/platformio.ini
+++ b/platformio.ini
@@ -26,7 +26,7 @@ build_flags =
 
 [common]
 lib_deps =
-    esphome/noise-c@0.1.3     ; api
+    esphome/noise-c@0.1.4     ; api
     makuna/NeoPixelBus@2.6.7  ; neopixelbus
 build_flags =
     -DESPHOME_LOG_LEVEL=ESPHOME_LOG_LEVEL_VERY_VERBOSE

